src/backend/lib/queue.js

```diff
@@ -25,7 +28,7 @@ function createQueue(name) {
         case 'subscriber':
```

Comment:
line 24 doesn't make sense for me. I know it is not your code. I am just trying to understand it. In the `const queue` I can't see who is going to pass the type to Bull constructor. What do you think?

Reply:
I am actually not sure how is it going to be passed either. What we would probably need to do is to pass it as a parameter to the _createQueue_ function, right?
```js
function createQueue(name, type) {
  const log = logger.child({ module: `queue:${name}` });
  const queue = new Bull(name, {
    createClient: type => {
      switch (type) {
        case 'client':
          return client;
        case 'subscriber':
          return subscriber;
        default:
          return new Redis(redisPort, redisHost, { password: redisPassword });
      }
    },
  })
// ...
return queue;
}
```

Reply:
essentially the Bull constructor here accepts a name and an object (supposed to be Redis instance?) like this:
```
Bull(name, {createClient: function, ?})
.on(...)
...
.on(...);
```
I looked into their documentation and I didn't find this signature. 
I found [this](https://github.com/OptimalBits/bull/issues/711) code similar to ours though (second comment).
I didn't find anything similar in their documentation.
I don't know. It needs more time. I will get back to this later or if you found something wrong, you may want to make a new issue.

Reply:
We looked into this in class, and I think what we have is good.  The type doesn't need to be passed in for the flow we use.